tests/ballerina-test-integration/src/test/java/org/ballerinalang/test/service/websocket/CustomHeaderClientSupportTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.test.service.websocket;
+
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.ballerinalang.test.context.ServerInstance;
+import org.ballerinalang.test.util.websocket.client.WebSocketTestClient;
+import org.ballerinalang.test.util.websocket.server.WebSocketRemoteServer;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.net.URISyntaxException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This Class tests receiving and sending of custom headers by Ballerina WebSocket client.
+ */
+public class CustomHeaderClientSupportTest extends WebSocketIntegrationTest {
+
+    private WebSocketRemoteServer remoteServer;
+    private ServerInstance ballerinaServerInstance;
+    private WebSocketTestClient client;
+    private static final String URL = "ws://localhost:9090/pingpong/ws";
+    private static final String RECEIVED_TEXT = "some-header-value";
+
+    @BeforeClass(description = "Initializes the Ballerina server with the ping_pong_and_client_custom_header.bal file")
+    public void setup() throws InterruptedException, BallerinaTestException, URISyntaxException {
+        remoteServer = new WebSocketRemoteServer(REMOTE_SERVER_PORT);
+        remoteServer.run();
+
+        String balPath = new File("src/test/resources/websocket/ping_pong_and_client_custom_header.bal")
+                .getAbsolutePath();
+        ballerinaServerInstance = ServerInstance.initBallerinaServer();
+        ballerinaServerInstance.startBallerinaServer(balPath);
+
+        client = new WebSocketTestClient(URL);
+        client.handshake();
```

Comment:
Shall we move the above code in setup to WebSocketIntegrationTest as it is used by several classes?